pkg/feature/api/segment.go

```diff
@@ -568,3 +622,62 @@ func (s *FeatureService) newSegmentListOrders(
 	}
 	return []*mysql.Order{mysql.NewOrder(column, direction)}, nil
 }
+
+func (s *FeatureService) injectFeaturesIntoSegments(
+	ctx context.Context,
+	segments []*featureproto.Segment,
+	featureIDsMap map[string][]string,
+	environmentNameSpace string,
+) error {
+	allFeatures, err := s.listAllFeatures(
```

Comment:
For the `ListSegments` API, getting all the feature flags in advance makes sense, but for the `GetSegment` API would be better to use the `GetFeatures` API to get only the target flags. But it would need two implementations, so I think we can go as it is, and if we see performance issues, we can refactor this part.